src/bin/wasm-tools/mutate.rs

```diff
@@ -75,17 +75,17 @@ impl Opts {
         };
 
         let stdout = stdout();
-        let (mut output, output_name): (Box<dyn Write>, _) = match &self.output {
+        let (mut output, output_name, output_wat): (Box<dyn Write>, _, _) = match &self.output {
             Some(f) => {
                 let output = Box::new(
                     fs::File::create(f)
                         .with_context(|| format!("failed to create '{}'", f.display()))?,
                 );
-                (output, f.display().to_string())
+                (output, f.display().to_string(), false)
             }
             None => {
                 let output = Box::new(stdout.lock());
-                (output, "<stdout>".to_string())
+                (output, "<stdout>".to_string(), true)
             }
         };
```

Comment:
Maybe it's just early in the morning, but it took me a second to understand what was happening here and why we were sometimes outputting WAT instead of Wasm. I think a little comment in the `stdout` match arm would be really helpful, something like:

> When we are writing to `stdout` and presumably a human is reading the output, don't emit Wasm, as that is not human readable. Instead, emit the WAT disassembly.

Although this does make me wonder whether this should be a dedicated `--wat` flag instead, since it means that piping `stdout` somewhere else might not work as expected. Thoughts?

Reply:
`--wat` seems reasonable to me!